lib/pbench/test/unit/agent/test_config.py

```diff
@@ -34,3 +38,72 @@ def test_results_attr():
         "pbench"
         in PbenchAgentConfig(os.environ["_PBENCH_AGENT_CONFIG"]).results["user"]
     )
+
+
+# Monkeypatching different Classes and Methods to test Pbench-run value
+
+
+class MockPbenchAgentConfig:
+    """Mock class for PbenchAgentConfig"""
+
+    def __init__(self, cfg_name: str):
+        self.pbench_run = "/test/pbench-run"
+        self.pbench_log = "logger"
+        self.pbench_install_dir = Path("/test/install_dir")
+        self.ssh_opts = "/ssh_opts"
+        self.scp_opts = "/scp_opts"
+
+
+def mock_exist(path: Path):
+    """mock method for Path.exist"""
+    return True
+
+
+def mock_current_context():
+    """mock get_current_context from click package"""
+    None
+
+
+class MockContext:
+    """Mocking Context"""
+
+    config = "/test/pbench-config-path"
+
+
+class TestCommand(BaseCommand):
+    """Mock of class that extends BaseCommand"""
+
+    def __init__(self):
+        super(TestCommand, self).__init__(MockContext())
+
+    def execute(self):
+        pass
+
+
+def test_pbench_run_path(monkeypatch):
+    """Use pbench_run value from pbench-agent.cfg file"""
+    assert os.environ.get("pbench_run") is None
+    monkeypatch.setattr(PbenchAgentConfig, "__init__", MockPbenchAgentConfig.__init__)
+    monkeypatch.setattr(Path, "exists", mock_exist)
+    pbench_run_val = TestCommand().pbench_run
+    assert str(pbench_run_val) == "/test/pbench-run"
+
+
+def test_pbench_run_as_env(monkeypatch):
+    """Use pbench_run value, passed manually through
+    `pbench-run` environment variable"""
+    monkeypatch.setenv("pbench_run", "/test/pbench-run/environ")
+    monkeypatch.setattr(PbenchAgentConfig, "__init__", MockPbenchAgentConfig.__init__)
+    monkeypatch.setattr(Path, "exists", mock_exist)
```

Comment:
Since all you want to do is return `True` (or, in the case of `test_pbench_run_dir_existence()`, `False`), you might want to consider using a lambda here instead of referencing a standalone function, since the expression is so simple:
```
    monkeypatch.setattr(Path, "exists", lambda path: True)
```
This saves the reader from having to refer elsewhere to determine what the mock does (and, it lets you choose locally whether to return `True` or `False`).